components/properties/src/props.rs

```diff
@@ -870,3 +879,31 @@ impl CanonicalCombiningClass {
     pub const DoubleAbove: CanonicalCombiningClass = CanonicalCombiningClass(234); // name="DA"
     pub const IotaSubscript: CanonicalCombiningClass = CanonicalCombiningClass(240); // name="IS"
 }
+
+//
+// String properties
+//
+
+/// String property Bidi_Mirroring_Glyph.
+///
+/// Informative mapping for substituting characters in an implementation of bidirectional mirroring.
+/// This maps a subset of characters with the Bidi_Mirrored property to other characters that normally
+/// are displayed with the corresponding mirrored glyph. When a character with the Bidi_Mirrored
+/// property has the default value for Bidi_Mirroring_Glyph, that means that no other character exists
+/// whose glyph is appropriate for character-based glyph mirroring.
+///
+/// See <https://www.unicode.org/reports/tr44/> .
+/// 
+/// ICU4X represents the property values as `Option<char>`, which is effectively equivalent
+/// to ICU's representation of a `u32` (in which  a code point with no mirroring glyph has
+/// a return value of the same input value).
+///
+/// Also see `u_charMirror` in ICU4C.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "datagen", derive(databake::Bake))]
+#[cfg_attr(feature = "datagen", databake(path = icu_properties))]
+#[allow(clippy::exhaustive_structs)] // newtype
+#[repr(transparent)]
+#[zerovec::make_ule(BidiMirroringGlyphULE)]
+pub struct BidiMirroringGlyph(pub u32);
```

Comment:
Issue: We should make this a 3-byte type so that the data provider also stores it as 3 bytes.

`[u8; 3]` is one way to accomplish this. Alternatively, a cleaner approach would be to add `UnvalidatedChar` in the zerovec crate with the correct semantics.

Reply:
I'd like to use the high 3 bits to fit more related properties in this data structure, so just "unvalidated char" wouldn't the the whole semantic. We might reuse the [`U24` struct](https://github.com/unicode-org/icu4x/commit/90db7c6a5ef3d64e969850c2655e8b9e45948c81#diff-8d43906dd2fed874a89265d4d6b07aa4f04c5a391fedbc66fb4b912251ed70b4) that the normalizer had before.

Quoting self from Slack:

> I do think it makes sense to bundle at least Bidi_Mirroring_Glyph, Bidi_Mirrored, Bidi_Paired_Bracket , and Bidi_Paired_Bracket_Type as a 24-bit value in one trie. I'm less sure of whether it makes sense to extend the trie value to 32 bits and bake Bidi_Class in the additional 8 bits.